src/components/profile-form.html

```diff
@@ -0,0 +1,87 @@
+<link rel="import" href="~@banno/polymer/polymer-element.js">
+<link rel="import" href="shared-styles.html">
+
+<dom-module id="profile-form">
+  <template>
+    <style include="shared-styles">
+      :host {
+        display: block;
+        padding: 10px;
+      }
+
+      input:not(:placeholder-shown):not(:focus):invalid {
+        border: 1px solid firebrick;
+        background: rgb(236, 172, 172);
+      }
+    </style>
+    <form class="validate">
+      <fieldset>
+        <legend id="editInfo">Your Profile:</legend>
+        <p>*required</p>
+        <div class="form-group">
+          <label for="firstName" id="inputFirstName">First Name *</label>
+          <input type="text" minlength="3" maxlength="12" class="form-control"
+            aria-describedby="createNewUser inputFirstName" placeholder="Enter first name"
+            value="{{userData.firstName::change}}" title="Please enter your first name." pattern="^[a-zA-Z]+$" required
+            autocomplete="fname">
+        </div>
+
+        <div class="form-group">
+          <label for="lastName" id="inputLastName">Last Name*</label>
+          <input type="text" minlength="3" maxlength="23" class="form-control" aria-describedby="createNewUser inputLastName"
+            placeholder="Enter last name" value="{{userData.lastName::change}}" title="Please enter your last name."
+            pattern="^[a-zA-Z]+$" required autocomplete="lname">
+        </div>
+
+        <div class="form-group">
+          <label for="phone" id="inputPhone">Phone </label>
+          <input type="tel" pattern="^(?:(?:\+?1\s*(?:[.-]\s*)?)?(?:\(\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*\)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\s*(?:[.-]\s*)?)?([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})(?:\s*(?:#|x\.?|ext\.?|extension)\s*(\d+))?$" class="form-control" aria-describedby="createNewUser inputPhone"
```

Comment:
_look at that regex_ 😎

I would just put it in a property rather than directly in the partial
```suggestion
          <input type="tel" pattern="phonePattern" class="form-control" aria-describedby="createNewUser inputPhone"
```

and then in the js

```
static get properties() {
  return {
    ...
    phonePattern: {
      type: String,
      value: '^(?:(?:\+?1\s*(?:[.-]\s*)?)?(?:\(\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*\)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\s*(?:[.-]\s*)?)?([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})(?:\s*(?:#|x\.?|ext\.?|extension)\s*(\d+))?$'
    }
  };
}
```

Reply:
cool. i was thinking there was probably some component imeplementation stuff I was missing - is there anything else you'd recommend - or is there somewhere on banno or elsewhere that is a good example of using polymer/components to do all of this in a particularly good way?

Reply:
i'm sure there is, just off the top of my head I'm not sure where in the code it'd be. you'd maybe want to try digging around in Settings or Dashboard. But this is close enough that I'm going to guess there isn't a _ton_ more you could do that'd get you closer to our own input validation.